SpatialGDK/Source/SpatialGDK/Private/Utils/ComponentFactory.cpp

```diff
@@ -50,13 +62,24 @@ bool ComponentFactory::FillSchemaObject(Schema_Object* ComponentObject, UObject*
 			const FRepParentCmd& Parent = Changes.RepLayout.Parents[Cmd.ParentIndex];
 
 #if TRACE_LIB_ACTIVE
-			if (LatencyTracer != nullptr)
+			if (LatencyTracer != nullptr && OutLatencyTraceId != nullptr)
 			{
-				OutLatencyTraceId = LatencyTracer->RetrievePendingTrace(Object, Cmd.Property);
-				if (OutLatencyTraceId == USpatialLatencyTracer::InvalidTraceKey)
+				TraceKey PropertyKey = InvalidTraceKey;
+				PropertyKey = LatencyTracer->RetrievePendingTrace(Object, Cmd.Property);
+				if (PropertyKey == InvalidTraceKey)
 				{
 					// Check for sending a nested property
-					OutLatencyTraceId = LatencyTracer->RetrievePendingTrace(Object, Parent.Property);
+					PropertyKey = LatencyTracer->RetrievePendingTrace(Object, Parent.Property);
+				}
+				if (PropertyKey != InvalidTraceKey)
+				{
+					// If we have already got a trace for this actor/component, we will end one of them here
+					if (*OutLatencyTraceId != InvalidTraceKey)
+					{
+						UE_LOG(LogSpatialNetSerialize, Warning, TEXT("%s property trace being dropped because too many active on this actor (%s)"), *Cmd.Property->GetName(), *Object->GetName());
```

Comment:
Use `LogComponentFactory`
Driveby, can you fix other misuses of this log type in this file?